File: Java/src/Problem006.java
/* Project Euler, prablem 6
 * The sum of the squares of the first ten natural numbers is,
 * 
 * 1^2 + 2^2 + ... + 10^2 = 385
 * The square of the sum of the first ten natural numbers is,
 * (1 + 2 + ... + 10)2 = 552 = 3025
 * 
 * Hence the difference between the sum of the squares of the first
 * ten natural numbers and the square of the sum is 3025 - 385 = 2640.
 * 
 * Find the difference between the sum of the squares of the first
 * one hundred natural numbers and the square of the sum.
 * 
 * not sure if this is a brute force solution or not, is fast
 * 
 * answer = 25164150;  //correct
 */

public class Problem006 {
	public static void main(String[] args){

		//double time = System.currentTimeMillis();
		
		int square = 0;
		int sum = 0;
		
		for (int i = 1; i <= 100; i++) {
			square += i * i;
			sum += i;
		}
		System.out.println((sum * sum) - square);
		//System.out.println(System.currentTimeMillis() - time);
	}
}

File: Ruby/Problem006.rb
=begin
The sum of the squares of the first ten natural numbers is,
    1^2 + 2^2 + ... + 10^2 = 385

The square of the sum of the first ten natural numbers is,
    (1 + 2 + ... + 10)^2 = 3025

Hence the difference between the sum of the squares of the 
first ten natural numbers and the square of the sum is,
    3025 - 385 = 2640

Find the difference between the sum of the squares of the 
first one hundred natural numbers and the square of the sum.
=end



def sumOfSq(num)
    sum = 0
    1.upto(num) { |i|
        sum += i ** 2
    }
    return sum
end

def sqOfSum(num)
    sq = 0
    1.upto(num) { |i|
        sq += i
    }
    return sq ** 2
end

puts sqOfSum(100) - sumOfSq(100)
